### PR TITLE
Update to github.com/golang-jwt/jwt/v5 package

### PR DESCRIPTION
**Pull request by Danielius1922**

### CODE REVIEW

File: coap-gateway/service/log.go [hunk 1]
```diff
@@ -188,7 +188,10 @@ func (c *session) msgToLogCoapMessage(req *pool.Message, withToken bool) logCoap
 	rq := coapgwMessage.ToJson(req, c.server.config.Log.DumpBody, withToken)
 	var sub string
 	if v, err := c.GetAuthorizationContext(); err == nil {
-		sub = v.GetJWTClaims().Subject()
+		subject, err := v.GetJWTClaims().GetSubject()
+		if err != nil {
```
Comment (jkralik) [inline coap-gateway/service/log.go]: `err == nil`

File: coap-gateway/service/auth.go [hunk 2]
```diff
@@ -38,18 +38,21 @@ func newAuthInterceptor() Interceptor {
...
 		return nil, err
 	}
-	return pkgJwt.Claims(m), nil
+	return pkgJwt5.Claims(m), nil
 }
 
-func (s *Service) VerifyDeviceID(tlsDeviceID string, claim pkgJwt.Claims) error {
-	jwtDeviceID := claim.DeviceID(s.config.APIs.COAP.Authorization.DeviceIDClaim)
+func (s *Service) verifyDeviceID(tlsDeviceID string, claim pkgJwt5.Claims) error {
+	jwtDeviceID, err := claim.GetDeviceID(s.config.APIs.COAP.Authorization.DeviceIDClaim)
```
Comment (jkralik) [inline coap-gateway/service/auth.go:52]: check for `if s.onfig.APIs.COAP.Authorization.DeviceIDClaim == ""` before `claim.GetDeviceID(...)`

File: pkg/errors/errors.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	for _, e := range we.errs {
+		if errors.Is(e, err) {
+			return true
+		}
+	}
+	return false
+}
+
+// Simple error wrapper that chains errors and prints them in a "%s(: %s)*" format
+func NewError(message string, err error, more ...error) error {
```
Comment (jkralik) [inline pkg/errors/errors.go]: pls use `github.com/hashicorp/go-multierror` if it is possible

File: pkg/security/jwt5/validator.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+func (v *Validator) ParseToken(token string) (*jwt.Token, error) {
+	if token == "" {
+		return nil, ErrMissingToken
+	}
+	return jwt.Parse(token, v.keys.GetOrFetchKey, jwt.WithIssuedAt())
+}
+
+func (v *Validator) Parse(token string) (jwt.MapClaims, error) {
+	t, err := v.ParseToken(token)
+	if t == nil {
```
Comment (jkralik) [inline pkg/security/jwt5/validator.go]: `err != nil`

File: pkg/security/jwt5/validator_test.go [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+	testJwt "github.com/plgd-dev/hub/v2/test/security/jwt5"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	token = `eyJhbGciOiJSUzI1NiIsImtpZCI6IjdmNzM1NWJmYmY0ZjVmOTNkZjJiZjg3OWE3OGUyMjNhIiwidHlwIjoiSldUIn0.eyJuYmYiOjE1NjM4ODk5NTgsImV4cCI6MTU2Mzg5MzU1OCwiaXNzIjoiaHR0cDovL2lkZW50aXR5LXNlcnZlcjozMDAxIiwiYXVkIjpbImh0dHA6Ly9pZGVudGl0eS1zZXJ2ZXI6MzAwMS9yZXNvdXJjZXMiLCJ0ZXN0LnJlc291cmNlIl0sImNsaWVudF9pZCI6InRlc3QuY2xpZW50LmlkIiwic3ViIjoiMWI4N2VmZmEtMzRlMi00YTQ0LTgyYzYtNmUwYWI4MDIwOWZmIiwiYXV0aF90aW1lIjoxNTYzODg5ODQ1LCJpZHAiOiJsb2NhbCIsImVtYWlsIjoidXNlckBleGFtcGxlLmNvbSIsInRlbmFudF9pZCI6IjdlYWE1MzJhLWMyYmYtNDRiNC05MTA5LWE3ODBmNGY2MzI2NCIsImRlZmF1bHRfcHJvZHVjdF91cmkiOiIvIiwicm9sZXMiOlsidGVzdC5yb2xlMSIsInRlc3Qucm9sZTIiXSwic2NvcGUiOlsidGVzdC5zY29wZSJdLCJhbXIiOlsicHdkIl19.jwgEJpn9aYZrWFzMRvW9ABZpA_MnZDNZcfWJtFm-luyYBm2D06P6bsKTH0mYcxKTVSVHnEczRecQumwtWjyvyVVtCLDOW1GVPseChw-By152QxOIeOpsJh2zPEKlVaqvPkFKOfFGPWYb99RN8-Cfv9hKYtUCrFhHCwcWcrzCngiYElGvRkyObBWyA7M2V_BWJGmO7W_j1e5TQF8PiD8ZEgNhnMd7jcEM0tGAH-v0aiV-X37gFq0bRkU3cb3xZo_4s_eLnTxfe270webbMo4-mIOFzClG7asdkgGQvNnO-aN9bvxBtii6aJVnYT5UIQDEofeJG-PdDVhkVyfSl__EHw`
```
Comment (jkralik) [inline pkg/security/jwt5/validator_test.go]: remove - pls generate if it is possible

File: pkg/errors/errors.go (hunk 3 above)
Comment (Danielius1922) [reply pkg/errors/errors.go]: It seems I don't need this wrapper at all, I want to error.ErrorIs to work in the case that I have a base error and I want to wrap it to add some additional information, but it seems `fmt.Errorf` works if I use `%w`.

File: coap-gateway/service/auth.go (hunk 2 above)
Comment (Danielius1922) [reply coap-gateway/service/auth.go:52]: `claim.GetDeviceID` returns `"", nil` in that case. Or do you mean to move

```go
if s.config.APIs.COAP.Authorization.DeviceIDClaim == "" {
  return nil
}
```
before the call? That will change the behaviour if `tlsDeviceID == ""`

File: pkg/security/jwt5/validator.go (hunk 4 above)
Comment (Danielius1922) [reply pkg/security/jwt5/validator.go]: this seems to be a deliberate behavior, at least some tests rely that you get the token even if you get an error. Do we want to modify it?

File: coap-gateway/service/log.go (hunk 1 above)
Comment (Danielius1922) [reply coap-gateway/service/log.go]: fixed

File: pkg/security/jwt5/validator_test.go (hunk 5 above)
Comment (Danielius1922) [reply pkg/security/jwt5/validator_test.go]: fixed

**Review by jkralik — APPROVED** [review]
👍

File: pkg/security/jwt/scopeClaims.go [hunk 6]
```diff
@@ -1,51 +1,86 @@
...
+	requiredScopes := make([]string, 0, len(scope))
 	for _, s := range scope {
-		requiredScopes = append(requiredScopes, regexp.MustCompile(regexp.QuoteMeta(s)))
+		requiredScopes = append(requiredScopes, regexp.MustCompile(regexp.QuoteMeta(s)).String())
 	}
 	return NewRegexpScopeClaims(requiredScopes...)
 }
 
-func NewRegexpScopeClaims(scope ...*regexp.Regexp) *ScopeClaims {
+func NewRegexpScopeClaims(scope ...string) *ScopeClaims {
```
Comment (jkralik) [inline pkg/security/jwt/scopeClaims.go]: why you change `*regexp.Regexp`  to `string` in *ScopeClaims ?

**Review by jkralik — APPROVED** [review]

**Pull request merged**